src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/service/ConfigurationSigningKeyServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ * <p>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * <p>
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * <p>
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.centralserver.restapi.service;
+
+import ee.ria.xroad.common.identifier.ClientId;
+import ee.ria.xroad.signer.protocol.dto.KeyInfo;
+import ee.ria.xroad.signer.protocol.dto.KeyUsageInfo;
+import ee.ria.xroad.signer.protocol.dto.TokenInfo;
+
+import lombok.RequiredArgsConstructor;
+import org.niis.xroad.centralserver.restapi.service.exception.NotFoundException;
+import org.niis.xroad.centralserver.restapi.service.exception.SignerProxyException;
+import org.niis.xroad.cs.admin.api.dto.ConfigurationSigningKey;
+import org.niis.xroad.cs.admin.api.dto.KeyLabel;
+import org.niis.xroad.cs.admin.api.facade.SignerProxyFacade;
+import org.niis.xroad.cs.admin.api.service.ConfigurationSigningKeyService;
+import org.niis.xroad.cs.admin.api.service.SystemParameterService;
+import org.niis.xroad.cs.admin.api.service.TokensService;
+import org.niis.xroad.cs.admin.core.entity.ConfigurationSigningKeyEntity;
+import org.niis.xroad.cs.admin.core.entity.ConfigurationSourceEntity;
+import org.niis.xroad.cs.admin.core.repository.ConfigurationSourceRepository;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.UUID;
+
+import static org.niis.xroad.centralserver.restapi.service.exception.ErrorMessage.CONFIGURATION_NOT_FOUND;
+import static org.niis.xroad.centralserver.restapi.service.exception.ErrorMessage.KEY_GENERATION_FAILED;
+import static org.niis.xroad.cs.admin.api.dto.PossibleAction.GENERATE_KEY;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class ConfigurationSigningKeyServiceImpl implements ConfigurationSigningKeyService {
```

Comment:
We're pretty much stepping on each other's toes with the given epic, so as a result it has pretty much become a race condition, but the given implementation and it's interface should be unified with the existing [ConfigurationSigningKeysServiceImpl](https://github.com/nordic-institute/X-Road/blob/develop-7.x/src/centralserver/admin-service/core/src/main/java/org/niis/xroad/centralserver/restapi/service/ConfigurationSigningKeysServiceImpl.java)

Reply:
Refactored.